plugins/modules/ndo_mac_sec_policy.py

```diff
@@ -0,0 +1,424 @@
+                            end=mso.verify_time_format(mac_sec_key.get("end_time")) if mac_sec_key.get("end_time") else None,
+                        )
+                    )
+
+                ops.append(dict(op="replace", path="{0}/{1}/macsecKeys".format(path, match.index), value=mac_sec_keys_list))
+                match.details["macsecKeys"] = mac_sec_keys
+            elif mac_sec_keys == []:
+                # remove mac_sec_keys if the list is empty
+                ops.append(dict(op="remove", path="{0}/{1}/macsecKeys".format(path, match.index)))
+                match.details.pop("macsecKeys")
+
+            mso.sanitize(match.details)
+
+        else:
+            mac_sec_param_map = {}
+
+            payload = {"name": mac_sec_policy, "templateId": mso_template.template.get("templateId"), "schemaId": mso_template.template.get("schemaId")}
+            payload["adminState"] = admin_state
+            payload["type"] = interface_type
+
+            if description:
+                payload["description"] = description
+
+            if cipher_suite:
+                mac_sec_param_map["cipherSuite"] = NDO_CIPHER_SUITE_MAP.get(cipher_suite)
+            if window_size:
+                mac_sec_param_map["windowSize"] = window_size
+            if security_policy:
+                mac_sec_param_map["securityPol"] = NDO_SECURITY_POLICY_MAP.get(security_policy)
+            if sak_expiry_time:
+                mac_sec_param_map["sakExpiryTime"] = sak_expiry_time
+
+            if interface_type == "access":
+                if confidentiality_offset:
+                    mac_sec_param_map["confOffSet"] = "offset" + str(confidentiality_offset)
+                if key_server_priority:
+                    mac_sec_param_map["keyServerPrio"] = key_server_priority
+
+                payload["macsecParams"] = mac_sec_param_map
+
+            mac_sec_keys_list = []
+            if mac_sec_keys:
+                for mac_sec_key in mac_sec_keys:
+                    mac_sec_key_dict = {
+                        "keyname": mac_sec_key.get("key_name"),
+                        "psk": mac_sec_key.get("psk"),
+                    }
+                    if mac_sec_key.get("start_time"):
+                        mac_sec_key_dict["start"] = mac_sec_key.get("start_time")
+                    if mac_sec_key.get("end_time"):
+                        mac_sec_key_dict["end"] = mac_sec_key.get("end_time")
+                    mac_sec_keys_list.append(mac_sec_key_dict)
+                payload["macsecKeys"] = mac_sec_keys_list
+
+            ops.append(dict(op="add", path="{0}/-".format(path), value=copy.deepcopy(payload)))
+
+            mso.sanitize(payload)
+
+        mso.existing = mso.proposed
+
+    elif state == "absent":
+        if match:
+            ops.append(dict(op="remove", path="{0}/{1}".format(path, match.index)))
+        mso.existing = {}
+
+    if not module.check_mode and ops:
```

Comment:
By referring to this module https://github.com/CiscoDevNet/ansible-mso/pull/531/files can you make changes here such that you capture the response from the patch API and populate the  mso.existing variable from it?

Reply:
changed